library/std/src/sys/unix/fd.rs

```diff
@@ -109,7 +109,7 @@ impl FileDesc {
                 self.as_raw_fd(),
                 buf.as_mut_ptr() as *mut c_void,
                 cmp::min(buf.len(), READ_LIMIT),
-                offset as i64,
+                offset as libc::off_t,
```

Comment:
The cast should be `offset as libc::off64_t` for Linux & Android, and `offset as libc::off_t` for everybody else.